src/geode/geometry/intersection.cpp

```diff
@@ -838,14 +839,172 @@ namespace geode
             }
             if( segment_intersections.empty() )
             {
-                return {};
+                return { IntersectionType::NONE };
             }
             return { std::move( segment_intersections ),
-                std::move( line_intersections.correctness ) };
+                std::move( line_intersections.correctness.value() ) };
         }
         return line_intersections.type;
     }
 
+    IntersectionResult< absl::InlinedVector< Point3D, 2 > >
+        triangle_circle_intersection(
+            const Triangle3D& triangle, const Circle& circle )
+    {
+        const auto plane = triangle.plane();
+        if( !plane )
+        {
+            return { IntersectionType::NONE };
+        }
+        const auto plane_intersection =
+            plane_circle_intersection( plane.value(), circle );
+        if( !plane_intersection.has_intersection() )
+        {
+            // The planes are parallel or nonintersecting.
+            return { plane_intersection.type };
+        }
+        absl::InlinedVector< Point3D, 2 > result;
+        CorrectnessInfo< absl::InlinedVector< Point3D, 2 > > correctness;
+        auto& intersections = plane_intersection.result.value();
+        for( const auto i : Indices{ intersections } )
+        {
+            auto& intersection = intersections[i];
+            auto triangle_output =
+                point_triangle_distance( intersection, triangle );
+            if( std::get< 0 >( triangle_output ) <= global_epsilon )
+            {
+                result.emplace_back( std::move( intersection ) );
+                correctness.first.second.emplace_back(
+                    std::move( std::get< 1 >( triangle_output ) ) );
+                correctness.second.second.emplace_back(
+                    plane_intersection.correctness->second.second[i] );
+            }
+        }
+        if( result.empty() )
+        {
+            return { IntersectionType::NONE };
+        }
+        correctness.first.first = true;
+        correctness.second.first = true;
```

Comment:
i do not understand?

Reply:
L887: all points are in the thick triangle because of test L874
L888: all points are in the thick circle or test on L861 will have already failed